src/dataset.js

```diff
@@ -205,6 +208,28 @@ function Dataset(bigQuery, id) {
 
   this.bigQuery = bigQuery;
 
+  /**
+   * @name Dataset#metadata
+   * @type {object}
+   */
+  this.metadata = options || {};
```

Comment:
Sorry to keep bringing this up. I think we need to completely separate `options` from `metadata`:

```js
this.metadata = {};

if (options.location) {
  this.metadata.location = options.location;
}
```

Reply:
Done.